l10n_it_fatturapa_pec/wizard/wizard_export_fatturapa.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Roberto Fichera - Level Prime Srl
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import base64
+import logging
+
+from odoo import models, _
+from odoo.exceptions import UserError
+
+from odoo.addons.l10n_it_fatturapa_out.wizard.wizard_export_fatturapa import (
+    fatturapaBDS
+)
+
+from odoo.addons.l10n_it_fatturapa.bindings.fatturapa import (
+    FatturaElettronica,
+    FatturaElettronicaBodyType
+)
+
+_logger = logging.getLogger(__name__)
+
+try:
+    from pyxb.exceptions_ import SimpleFacetValueError, SimpleTypeValueError
+except ImportError as err:
+    _logger.debug(err)
+
+
+class WizardExportFatturapa(models.TransientModel):
+    _inherit = "wizard.export.fatturapa"
+
+    def updateAttachment(self, attach, fatturapa):
+        attach_str = fatturapa.toxml(
+            encoding="UTF-8",
+            bds=fatturapaBDS,
+        )
+        fatturapaBDS.reset()
+        attach.write({
+            'datas': base64.encodestring(attach_str),
+        })
+
+    def reSetProgressivoInvio(self, attach, fatturapa):
+        # Xml file name uses the format VAT_XXXXX.xml and we are interested
+        # to get XXXXX
+        file_id = attach.name.split('_')[1].split('.')[0]
+        try:
+            fatturapa.FatturaElettronicaHeader.DatiTrasmissione.\
+                ProgressivoInvio = file_id
+        except (SimpleFacetValueError, SimpleTypeValueError) as e:
+            msg = _(
+                'FatturaElettronicaHeader.DatiTrasmissione.'
+                'ProgressivoInvio:\n%s'
+            ) % unicode(e)
+            raise UserError(msg)
+        return file_id
+
+    def exportFatturaPARegenerate(self):
```

Comment:
@robyf70 secondo te  vale la pena estrarre un metodo da `exportFatturaPA` da riutilizzare sia là che qua, in modo da ridurre la duplicazione del codice?

Reply:
@eLBati Probabilmente andrebbe splittato per dare possibilità di cambiare o aggiungere features

Reply:
Anche perchè, nell'implementazione attuale, se ci fosse qualcosa da correggere/modificare in `exportFatturaPA` , dovremmo ricordarci di correggere anche `exportFatturaPARegenerate`

Reply:
@eLBati Probabilmente andrebbe spostato fuori il blocco di generazione dell'XML di una fattura ed isolare la logica attorno per creare o rigenerare un XML completo.